engine/src/flutter/display_list/geometry/dl_path.cc

```diff
@@ -252,26 +268,90 @@ DlPath DlPath::operator+(const DlPath& other) const {
   return DlPath(path);
 }
 
-SkPath DlPath::ConvertToSkiaPath(const Path& path, const DlPoint& shift) {
-  SkPath sk_path;
-  sk_path.setFillType(ToSkFillType(path.GetFillType()));
+static void ReduceConic(DlPathReceiver& receiver,
+                        const DlPoint& p1,
+                        const DlPoint& cp,
+                        const DlPoint& p2,
+                        DlScalar weight) {
+  // We might eventually have conic conversion math that deals with
+  // degenerate conics gracefully (or have all receivers just handle
+  // them directly). But, until then, we will just convert them to a
+  // pair of quads and accept the results as "close enough".
+  if (p1 != cp) {
+    if (cp != p2) {
+      std::array<DlPoint, 5> points;
+      impeller::ConicPathComponent conic(p1, cp, p2, weight);
+      conic.SubdivideToQuadraticPoints(points);
+      receiver.QuadTo(points[1], points[2]);
+      receiver.QuadTo(points[3], points[4]);
+    } else {
+      receiver.LineTo(cp);
+    }
+  } else if (cp != p2) {
+    receiver.LineTo(p2);
+  }
+}
+
+class SkiaPathReceiver final : public virtual DlPathReceiver {
```

Comment:
This needs to be in an anonymous namespace too to avoid linking collisions.  Same for the other classes.

Reply:
Won't an anonymous namespace make naming collisions more likely?

Reply:
No, it makes it impossible.  It would be against the c++ spec for 2 different things to collide in different compilation units inside anonymous namespaces.  You can think of it as generating a namespace that is guaranteed to be unique for that cu.

Reply:
TIL